spec/models/sales_log_derived_fields_spec.rb

```diff
@@ -0,0 +1,164 @@
+require "rails_helper"
+require "shared/shared_examples_for_derived_fields"
+
+RSpec.describe SalesLog, type: :model do
```

Comment:
This should be the same set of tests as have been removed from `sales_log_spec` (with adjustments where I've actually changed the behaviour in this PR).

I had to make some adjustments in order to have them work without a date fixed in the past (which I needed for the tests relevant to this PR), and it ended up making sense to pull them out and adjust to avoid being too database heavy / having fixed time issues / relying on unspoken properties from the factory. This means they are now explicitly testing the `set_derived_fields!` function rather than it's actions when we save an update (which makes sense because this set of tests is clearly designed as unit tests).

As a bonus, this set runs significantly faster (on my machine at least).

Reply:
super! yes I think I wrote those old tests. this is much nicer building and then calling a method in memory. I'm not surprised it should be much faster. Every old test was making a db call for insert and then update as well.